components/camel-huawei/camel-huaweicloud-obs/src/main/java/org/apache/camel/component/huaweicloud/obs/models/OBSRegion.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.huaweicloud.obs.models;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.camel.util.ObjectHelper;
+
+/**
+ * Class containing Huawei Cloud OBS regions and endpoints
+ */
+public final class OBSRegion {
```

Comment:
Is an ENUM not a better class for this? Then you can also use Camel's enum support with automatic type conversion and avoid any mixed case problems, eg CN-NORTH-1 vs cn-north-1

Reply:
Thanks @davsclaus for the suggestion. For OBS alone, we use [a different cloud SDK](https://github.com/huaweicloud/huaweicloud-sdk-java-obs) whereas for other huawei cloud components, we have used [huaweicloud-sdk-java-v3](https://github.com/huaweicloud/huaweicloud-sdk-java-v3) . To be consistent with the [v3 sdk style of maintaining endpoints](https://github.com/huaweicloud/huaweicloud-sdk-java-v3/blob/master/services/functiongraph/src/main/java/com/huaweicloud/sdk/functiongraph/v2/region/FunctionGraphRegion.java), we decided to keep the similar pattern untill v3 team comes up with support for OBS.

Considering the potential mixed case problem, I have made it case insensitive to shield against different user behaviors in [this commit](https://github.com/apache/camel/pull/5753/commits/238b61819f1e86978b8abf5851b54b20a24b2883)

Let me know what you feel about this change.